src/swtpm/common.c

```diff
@@ -620,16 +633,30 @@ parse_tpmstate_options(char *options, char **tpmstatedir, mode_t *mode)
     }
 
     directory = option_get_string(ovs, "dir", NULL);
-    if (!directory) {
-        logprintf(STDERR_FILENO,
-                  "The file parameter is required for the tpmstate option.\n");
-        goto error;
-    }
     *mode = option_get_mode_t(ovs, "mode", 0640);
+    backend = option_get_string(ovs, "backend", NULL);
+    backend_uri = option_get_string(ovs, "backend_uri", NULL);
 
-    *tpmstatedir = strdup(directory);
-    if (!*tpmstatedir) {
-        logprintf(STDERR_FILENO, "Out of memory.");
+    if (directory) {
+        *tpmstatedir = strdup(directory);
+        if (!*tpmstatedir) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
+    } else if (backend && backend_uri) {
+        *tpmstatebackend = strdup(backend);
+        if (!*tpmstatebackend) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
+        *tpmstatebackend_uri = strdup(backend_uri);
+        if (!*tpmstatebackend_uri) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
+    } else {
+        logprintf(STDERR_FILENO,
+                  "The file parameter is required for the tpmstate option.\n");
```

Comment:
it's not the file parameter but the directory or backend+backenduri parameters

Reply:
Thanks, I'll fix the error message.